File: Data Structures & Algorithms/replace-elements-with-greatest-element-on-right-side/submission-0.java
class Solution {
    public int[] replaceElements(int[] arr) {
        int[] ans = new int[arr.length];
        int maxy = -1;
        for(int i = arr.length - 1; i >= 0; i--)
        {
            ans[i] = maxy;
            maxy = Math.max(arr[i],maxy);
        }
        return ans;
    }
}

File: Data Structures & Algorithms/find-lucky-integer-in-an-array/submission-0.cpp
class Solution {
public:
    int findLucky(vector<int>& arr) {
        vector<int>fr(501,0);
        for(int i : arr) fr[i]+=1;
        int ans = 0;
        for(int i : arr) if(i == fr[i]) ans = max(ans,i);
        return ans > 0 ? ans : -1;
    }
};